app/src/main/java/ch/epfl/culturequest/notifications/PushNotification.java

```diff
@@ -108,23 +113,31 @@ public Notification buildNotification(Context context) {
      * @return the pending intent
      */
     public PendingIntent selectPendingIntent(Context context, String channelId) {
-        Intent default_intent = new Intent(context, NavigationActivity.class);
-        default_intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-        PendingIntent default_pending_intent = PendingIntent.getActivity(context, 0, default_intent, PendingIntent.FLAG_IMMUTABLE);
-        
+        Intent intent;
         switch (channelId) {
+            // opens the profile of the user who sent the notification
             case FollowNotification.CHANNEL_ID:
-                return FollowNotification.getPendingIntent(context);
-            case ScanNotification.CHANNEL_ID:
-                return ScanNotification.getPendingIntent(context);
+                intent = new Intent(context, DisplayUserProfileActivity.class);
+                intent.putExtra("uid", senderId);
+                intent.putExtra("redirect", "home");
+                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+                return PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_MUTABLE);
+            // opens the profile of the current user
             case LikeNotification.CHANNEL_ID:
-                return LikeNotification.getPendingIntent(context);
-            case CompetitionNotification.CHANNEL_ID:
-                return CompetitionNotification.getPendingIntent(context);
-            case SightseeingNotification.CHANNEL_ID:
-                return SightseeingNotification.getPendingIntent(context);
+                intent = new Intent(context, NavigationActivity.class);
+                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+                intent.putExtra("redirect", "profile");
+                return PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_MUTABLE);
```

Comment:
These 2 lines appear a few time, would it be possible to limit that?

Reply:
Yes I will do that